docs/architecture/synthesis-superior-assistant-spec.md

```
@@ -0,0 +1,492 @@
+
+interface DynamicTraitsProvider {
+  getEffectiveTraits(base: AssistantTraits, context: TurnContextInput): Promise<AssistantTraits>;
+  proposeOverlay(evidence: TraitEvidence): Promise<DynamicTraitOverlay>;
+  approveOverlay(overlayId: string): Promise<void>;
+  rejectOverlay(overlayId: string): Promise<void>;
+}
+```
+
+### 5.3 Composition at Turn-Context Assembly Time
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)